dask_planner/src/expression.rs

```diff
@@ -471,8 +471,15 @@ impl PyExpr {
 
 /// Create a [DFField] representing an [Expr], given an input [LogicalPlan] to resolve against
 pub fn expr_to_field(expr: &Expr, input_plan: &LogicalPlan) -> Result<DFField> {
-    // TODO this is not the implementation that we really want and will be improved
-    // once some changes are made in DataFusion
-    let fields = exprlist_to_fields(&[expr.clone()], &input_plan.schema())?;
-    Ok(fields[0].clone())
+    match expr {
+        Expr::Sort { expr, .. } => {
+            // DataFusion does not support create_name for sort expressions (since they never
+            // appear in projections) so we just delegate to the contained expression instead
+            expr_to_field(expr, input_plan)
+        }
+        _ => {
+            let fields = exprlist_to_fields(&[expr.clone()], &input_plan)?;
```

Comment:
Note that with the latest DataFusion code, we now pass a `LogicalPlan` here instead of a schema.